lms/djangoapps/student_account/views.py

```diff
@@ -337,26 +253,24 @@ def _third_party_auth_context(request, redirect_to, tpa_hint=None):
     }
 
     if third_party_auth.is_enabled():
-        enterprise_customer = enterprise_customer_for_request(request)
-        if not enterprise_customer:
-            for enabled in third_party_auth.provider.Registry.displayed_for_login(tpa_hint=tpa_hint):
-                info = {
-                    "id": enabled.provider_id,
-                    "name": enabled.name,
-                    "iconClass": enabled.icon_class or None,
-                    "iconImage": enabled.icon_image.url if enabled.icon_image else None,
-                    "loginUrl": pipeline.get_login_url(
-                        enabled.provider_id,
-                        pipeline.AUTH_ENTRY_LOGIN,
-                        redirect_url=redirect_to,
-                    ),
-                    "registerUrl": pipeline.get_login_url(
-                        enabled.provider_id,
-                        pipeline.AUTH_ENTRY_REGISTER,
-                        redirect_url=redirect_to,
-                    ),
-                }
-                context["providers" if not enabled.secondary else "secondaryProviders"].append(info)
+        for enabled in third_party_auth.provider.Registry.displayed_for_login(tpa_hint=tpa_hint):
```

Comment:
Don't we still need the `if not enterprise_customer` guard around this?

Reply:
@georgebabey we have moved the enterprise related check in `enterprise_support` `utils` method `update_third_party_auth_context_for_enterprise`. There we are setting the `providers` and `secondaryProviders` to empty list in case if enterprise_customer is available.
```python
    if enterprise_customer:
        context['data']['third_party_auth']['providers'] = []
        context['data']['third_party_auth']['secondaryProviders'] = []
```

Reply:
Awesome, thanks for clarifying!